### PR TITLE
feat(rules): implement prompt-engineering validator (PE-001 to PE-004)

### LINKED ISSUE #40: Implement prompt-engineering validator: PE-001 to PE-004
## Summary
Implement prompt engineering validation rules. Rule IDs are defined in SARIF output but no validator exists.

## Rules to Implement
- **PE-001**: Repetition without emphasis or XML isolation
- **PE-002**: Chain-of-thought on simple task
- **PE-003**: Weak imperative language in critical rules
- **PE-004**: Ambiguous instructions

## Deliverables
- [ ] Create `crates/agnix-core/src/rules/prompt.rs`
- [ ] Add prompt file detection in lib.rs
- [ ] Add test fixtures in `tests/fixtures/prompt/`
- [ ] Tests for each rule

## References
- knowledge-base/VALIDATION-RULES.md

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Implements prompt engineering validation rules PE-001 through PE-004, addressing issue #40.

### Rules Implemented

| Rule | Severity | Description |
|------|----------|-------------|
| PE-001 | ERROR | Critical content in middle 40-60% zone ("lost in the middle") |
| PE-002 | ERROR | Chain-of-thought phrases in simple tasks |
| PE-003 | WARNING | Weak imperative language in critical sections |
| PE-004 | WARNING | Ambiguous instructions without specific criteria |

### Changes

- **New files:**
  - `crates/agnix-core/src/rules/prompt.rs` - PromptValidator implementation
  - `crates/agnix-core/src/schemas/prompt.rs` - Detection functions using OnceLock<Regex>
  - `tests/fixtures/prompt/*.md` - 8 test fixtures

- **Modified files:**
  - `crates/agnix-core/src/config.rs` - Added `prompt_engineering` category toggle
  - `crates/agnix-core/src/lib.rs` - Registered PromptValidator for Skill, ClaudeMd, GenericMarkdown
  - `crates/agnix-core/src/rules/mod.rs` - Export prompt module
  - `crates/agnix-core/src/schemas/mod.rs` - Export prompt module
  - `README.md`, `SPEC.md`, `CHANGELOG.md` - Documentation updates

### Test Plan

- [x] All 483 tests passing (`cargo test`)
- [x] Build passes (`cargo build --release`)
- [x] Clippy clean (`cargo clippy`)
- [x] Integration tests for each PE rule
- [x] Edge case tests (empty content, boundary conditions, config disabling)

## Closes #40

### CODE REVIEW

File: crates/agnix-core/src/schemas/prompt.rs [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+    if simple_tasks.is_empty() {
+        return results;
+    }
+
+    // Find CoT phrases in the same document
+    for (line_num, line) in content.lines().enumerate() {
+        if let Some(mat) = cot_pattern.find(line) {
+            // Report CoT found in a simple-task context
+            if let Some((_, task)) = simple_tasks.first() {
+                results.push(CotOnSimpleTask {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/prompt.rs]: Fixed in commit 036cdee. Added 5-line proximity check so CoT phrases are only flagged when within proximity of simple task indicators.

File: crates/agnix-core/src/schemas/prompt.rs [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+/// "if possible" create ambiguity about when the instruction applies.
+pub fn find_ambiguous_instructions(content: &str) -> Vec<AmbiguousInstruction> {
+    let mut results = Vec::new();
+    let pattern = ambiguous_term_pattern();
+
+    for (line_num, line) in content.lines().enumerate() {
+        // Skip lines that are clearly code blocks or examples
+        if line.trim_start().starts_with("```")
+            || line.trim_start().starts_with("//")
+            || line.trim_start().starts_with("#!")
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/prompt.rs]: Fixed in commit 036cdee. Added in_code_block state tracking to properly skip entire fenced code block contents.

File: crates/agnix-core/src/schemas/prompt.rs [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+                    column: mat.start(),
+                    phrase: mat.as_str().to_string(),
+                    task_indicator: task.clone(),
+                });
+            }
+        }
+    }
+
+    results
+}
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/prompt.rs:159]: Fixed in commit 036cdee. Added 5-line proximity check to only flag CoT phrases when they're within proximity of simple task indicators.

File: crates/agnix-core/src/schemas/prompt.rs [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+            || line.trim_start().starts_with("#!")
+        {
+            continue;
+        }
+
+        for mat in pattern.find_iter(line) {
+            // Extract some context around the match
+            let start = mat.start().saturating_sub(20);
+            let end = (mat.end() + 20).min(line.len());
+            let context = line[start..end].to_string();
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/prompt.rs]: Fixed in commit 88a122c. Implemented UTF-8 safe slicing using char_indices() with take_while/find pattern to prevent panics on multi-byte characters. Also added in_code_block state tracking.

File: crates/agnix-core/src/schemas/prompt.rs [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+    WEAK_LANGUAGE_PATTERN.get_or_init(|| {
+        Regex::new(r"(?i)\b(should|try\s+to|consider|maybe|might|could|possibly|preferably|ideally|optionally)\b")
+            .unwrap()
+    })
+}
+
+fn critical_section_pattern() -> &'static Regex {
+    CRITICAL_SECTION_PATTERN.get_or_init(|| {
+        Regex::new(r"(?i)^#+\s*.*(critical|important|required|mandatory|rules|must|essential|security|danger)")
+            .unwrap()
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/schemas/prompt.rs]: Fixed in commit 036cdee. Added word boundaries (\b) to critical_section_pattern regex to prevent matching substrings like 'Hypercritical'.
